packages/cosmic-swingset/src/agcc-agvm.js

```diff
@@ -0,0 +1,144 @@
+/* global process */
+// @ts-check
+import fs from 'fs';
+import { Worker } from 'worker_threads';
+import stringify from './helpers/json-stable-stringify.js';
+import { makeJsonRpcService } from './helpers/input-worker.js';
+
+const DAEMON_PORT = 321;
+
+const debug = (..._args) => {}; // console.error(..._args);
+
+const getFdsFromSpec = spec => {
+  const obj = JSON.parse(spec);
+
+  const { agd: { fd: fds = [] } = {} } = obj || {};
+  if (!Array.isArray(fds)) {
+    throw Error(
+      `VM spec ${spec} does not have an array for agd.fd, has ${fds}`,
+    );
+  }
+  const ifd = fds[0] || 0;
+  const ofd = fds[1] || ifd || 1;
+
+  return [ifd, ofd];
+};
+
+export const makeAgccFromSpec = spec => {
+  const [ifd, ofd] = getFdsFromSpec(spec);
```

Comment:
A shape like `{ agd: { fd: [inputFdNumber = 0, outputFdNumber = inputFdNumber] } }` strikes me as too complicated, especially when we control both sides and can refactor if/when needed.
```suggestion
/**
 * Construct an object representing the JavaScript side of a connection with an
 * Agoric blockchain node that will be a source of inbound VM messages on file
 * descriptor inputFd and will read responses on outputFd.
 *
 * @param {number} inputFd
 * @param {number} outputFd
 */
export const makeAgcc = (inputFd, outputFd) => {
```